internal/configs/resource.go

```diff
@@ -647,6 +576,210 @@ func decodeReplaceTriggeredBy(expr hcl.Expression) ([]hcl.Expression, hcl.Diagno
+					// This is no longer allowed and will be handled below
+					r.Managed.IgnoreChanges = append(r.Managed.IgnoreChanges, hcl.Traversal{hcl.TraverseAttr{Name: "*", SrcRange: expr.Range()}})
+					continue
+				}
+
+				// Might be a quoted string, sub in a parsed traversable expression
+				shimExpr, shimDiags := shimTraversalInString(expr, false)
+				diags = append(diags, shimDiags...)
+				if shimDiags.HasErrors() {
+					continue
+				}
+
+				traversal, travDiags := hcl.RelTraversalForExpr(shimExpr)
+				diags = append(diags, travDiags...)
+				if travDiags.HasErrors() {
+					continue
+				}
+				r.Managed.IgnoreChanges = append(r.Managed.IgnoreChanges, traversal)
+			}
+		} else {
+			// Invalid Expression
+			return diags.Extend(valDiags)
+		}
+	} else {
+		if val.Type().IsListType() || val.Type().IsSetType() || val.Type().IsTupleType() {
+			for _, item := range val.AsValueSlice() {
+				if item.Type() != cty.String {
+					diags = diags.Append(&hcl.Diagnostic{
+						Severity: hcl.DiagError,
+						Summary:  "Invalid ignore_changes input",
+						Detail:   fmt.Sprintf(`Expected collection of string, not %s`, item.Type().FriendlyName()),
+					})
+					continue
+				}
+
+				if item.AsString() == "*" {
+					r.Managed.IgnoreChanges = append(r.Managed.IgnoreChanges, hcl.Traversal{hcl.TraverseAttr{Name: "*", SrcRange: attr.Expr.Range()}})
+					continue
+				}
+				traversal, tDiags := hclsyntax.ParseTraversalAbs(
+					[]byte(item.AsString()),
+					"",
+					hcl.InitialPos,
+				)
+				diags = append(diags, tDiags...)
+				if tDiags.HasErrors() {
+					continue
+				}
+				traversal[0] = hcl.TraverseAttr{
+					Name:     traversal[0].(hcl.TraverseRoot).Name,
+					SrcRange: attr.Expr.Range(),
+				}
+				r.Managed.IgnoreChanges = append(r.Managed.IgnoreChanges, traversal)
+			}
+		} else if val.Type() == cty.String {
+			if val.AsString() == "all" {
+				r.Managed.IgnoreAllChanges = true
+			} else {
+				diags = append(diags, &hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  "Invalid ignore_changes ruleset",
+					Detail:   fmt.Sprintf("Expected \"all\" or [items], not %s.", val.AsString()),
+					Subject:  attr.Expr.Range().Ptr(),
+				})
+			}
+			return diags
+		} else {
+			diags = append(diags, &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "Invalid ignore_changes ruleset",
+				Detail:   fmt.Sprintf("Unexpected type %s", val.Type().FriendlyName()),
+				Subject:  attr.Expr.Range().Ptr(),
+			})
+
+		}
+	}
+
+	var ignoreAllRange hcl.Range
+	for _, trav := range r.Managed.IgnoreChanges {
+		if root, ok := trav[0].(hcl.TraverseAttr); ok && root.Name == "*" {
+			ignoreAllRange = trav.SourceRange()
+			diags = append(diags, &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "Invalid ignore_changes wildcard",
+				Detail:   "The [\"*\"] form of ignore_changes wildcard is was deprecated and is now invalid. Use \"ignore_changes = all\" to ignore changes to all attributes.",
```

Comment:
```suggestion
				Detail:   "The [\"*\"] form of ignore_changes wildcard is deprecated and now invalid. Please use \"ignore_changes = all\" to ignore changes to all attributes.",
```

Reply:
I'd like to keep this message consistent as it was simply moved from the existing code.